mvlearn/embed/kcca.py

```diff
@@ -167,266 +218,85 @@ def fit_transform(self, Xs):
-            idx[dnum] = False
-            proj = ccomp[idx > 0].mean(0)
-            pred = np.dot(iws[dnum], proj.T).T
-            pred = np.nan_to_num(_zscore(pred))
-            self.vpreds_.append(pred)
-            cs = np.nan_to_num(_rowcorr(vdata[dnum].T, pred.T))
-            self.vcorrs_.append(cs)
+        comp1 = []
+        comp2 = []
 
-        return self.vcorrs_
+        for i in range(weight1.shape[1]):
+            comp1.append(self.Kx@weight1[:, i])
+            comp2.append(self.Ky@weight2[:, i])
 
+        comp1 = np.transpose(np.asarray([l*(-10**18) for l in comp1]))
+        comp2 = np.transpose(np.asarray([l*10**18 for l in comp2]))
 
-def kcca(
-    Xs, reg=0.0, n_components=None,
-    ktype="linear", sigma=1.0, degree=2
-):
-
-    """
-    Sets up and solves the kernel CCA eigenproblem
-
-    Parameters
-    ----------
-    Xs : list of array-likes
-        - Xs shape: (n_views,)
-        - Xs[i] shape: (n_samples, n_features_i)
-        The data for kcca to fit to.
-        Each sample will receive its own embedding.
-
-
-    Returns
-    -------
-    comp : list of array-likes
-           Component to determine the canonical weights
-    """
+        self.components_ = [comp1, comp2]
 
-    kernel = [
-        _make_kernel(d, ktype=ktype, sigma=sigma,
-                     degree=degree) for d in Xs
-    ]
-
-    nDs = len(kernel)
-    nFs = [k.shape[0] for k in kernel]
-    n_components = (min([k.shape[1] for k in kernel])
-                    if n_components is None else n_components)
-
-    # Get the auto- and cross-covariance matrices
-    crosscovs = [np.dot(ki, kj.T) for ki in kernel for kj in kernel]
-
-    # Allocate left-hand side (LH) and right-hand side (RH):
-    LH = np.zeros((sum(nFs), sum(nFs)))
-    RH = np.zeros((sum(nFs), sum(nFs)))
-
-    # Fill the left and right sides of the eigenvalue problem
-    for i in range(nDs):
-        RH[
-            sum(nFs[:i]): sum(nFs[: i + 1]), sum(nFs[:i]): sum(nFs[: i + 1])
-        ] = crosscovs[i * (nDs + 1)] + reg * np.eye(nFs[i])
-
-        for j in range(nDs):
-            if i != j:
-                LH[
-                    sum(nFs[:j]): sum(nFs[: j + 1]),
-                    sum(nFs[:i]): sum(nFs[: i + 1])
-                ] = crosscovs[nDs * j + i]
-
-    LH = (LH + LH.T) / 2.0
-    RH = (RH + RH.T) / 2.0
-
-    maxCC = LH.shape[0]
-    r, Vs = eigh(LH, RH, eigvals=(maxCC - n_components, maxCC - 1))
-    r[np.isnan(r)] = 0
-    rindex = np.argsort(r)[::-1]
-    comp = []
-    Vs = Vs[:, rindex]
-    for i in range(nDs):
-        comp.append(Vs[sum(nFs[:i]): sum(nFs[: i + 1]), :n_components])
-    return comp
-
-
-def _zscore(d):
-    """
-    Calculates z-score of data
-
-    Parameters
-    ----------
-    d : array
-        Data of interest
-
-    Returns
-    -------
-    z : array
-        Z-score
-    """
-    z = (d - d.mean(0)) / d.std(0)
-    return z
+        return self
 
+    def fit_transform(self, Xs):
```

Comment:
Since this inherits from BaseEmbed, you shouldn't have to redefine fit_transform, right?

Reply:
shouldn't need a fit_transform since that method was instantiated in the BaseEmbed and this doesn't change anything.